components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/validators/TokenValidationHandler.java

```diff
@@ -753,4 +760,13 @@ private void addAllowedScopes(OAuth2TokenValidationMessageContext oAuth2TokenVal
         String[] scopesToReturn = (String[]) ArrayUtils.addAll(scopes, allowedScopes);
         oAuth2TokenValidationMessageContext.getResponseDTO().setScope(scopesToReturn);
     }
+
+    private void addAllowedScopes(OAuth2IntrospectionResponseDTO introResp, AccessTokenDO accessTokenDO,
+                                  String[] allowedScopes) {
```

Comment:
change the allowedScopes -> requestedAllowedScopes

Reply:
fixed with f5642b437c1e3a57b5c669dbc18b32d53d613f68